### ISSUE TITLE
Import time: `scipy.signal`, `torchvision` and `matplotlib` are imported eagerly by `import torchmetrics`

### ISSUE THREAD

**Issue by frazane** [Lightning-AI/torchmetrics#3457]
## 🚀 Feature

Don't import optional dependencies (`scipy.signal`, `torchvision`, `matplotlib`) at `import torchmetrics` time just because they happen to be installed.

### Motivation

`python -X importtime -c "import torchmetrics"` on 1.9.0 (torch 2.10, Python 3.12, warm cache), cumulative:

```
torchmetrics                   3.30s
  torch                        1.22s
  scipy.signal                 0.66s
  torchvision                  0.85s
  torchmetrics.utilities.plot  0.31s
```

Around 1.8 s of the 2.1 s that is not torch goes to three optional packages that nothing in the import path uses. Lightning pays it too, since `lightning_fabric/utilities/imports.py` calls `compare_version("torchmetrics", ...)` and that does `importlib.import_module`.

Sources, verified on `master` and 1.9.0:

`scipy.signal` comes from the SRMRpy `hamming` back-compat patch, which sits in `torchmetrics/__init__.py`, `torchmetrics/audio/__init__.py` and `torchmetrics/functional/audio/__init__.py`. All three run eagerly, so all three have to move. `srmrpy` appears only in `requirements/audio_test.txt` and `tests/unittests/audio/test_srmr.py`; the library never imports it.

`torchvision` comes from `functional/image/arniqa.py`, which imports `transforms` and `models.resnet50` at module level under `if _TORCHVISION_AVAILABLE:`. Already tracked in #3314, with open PR #3432; mentioned here only for completeness.

`matplotlib` comes from `utilities/plot.py`, which imports `matplotlib`, `matplotlib.axes` and `matplotlib.pyplot` under `if _MATPLOTLIB_AVAILABLE:`. Around 140 metric modules import it for `_AX_TYPE`, `_PLOT_OUT_TYPE`, `_CMAP_TYPE` and `style_change`.

### Pitch

Move the imports to first use, as `functional/audio/srmr.py` and `functional/image/lpips.py` already do:

- Drop the `scipy.signal` patch from the three `__init__.py` files; apply it in the audio tests or in the `Lightning-Sandbox/SRMRpy` fork if it is still needed.
- `utilities/plot.py`: make the type aliases `TYPE_CHECKING`-only and resolve `plt.style.context` inside `style_change`.
- `arniqa`: #3432.

### Alternatives

A module-level `__getattr__` (PEP 562) over the domain re-exports would cover this wholesale, at the cost of static analysis.

### Additional context

Follows #2171 (closed by #2181, #2182, #2215, #2216, #2316), which did this for the audio/detection/image/text dependencies; these paths were missed or added later.

**Comment by VenishPaneliya** [comment]
I'd like to take this on.

Scope: the `scipy.signal` and `matplotlib` halves only — `torchvision`/arniqa is already covered by #3432, so I'll leave it alone to avoid conflicting.

Plan:

1. **`scipy.signal`** — drop the SRMRpy `hamming` back-compat patch from the three eager `__init__.py` files. `srmrpy` is only used by `tests/unittests/audio/test_srmr.py`, so the patch moves into the test path where it's actually needed.
2. **`matplotlib`** — in `utilities/plot.py`, define `_AX_TYPE`/`_PLOT_OUT_TYPE`/`_CMAP_TYPE` under `TYPE_CHECKING` with `object` fallbacks at runtime, and make `style_change` import `matplotlib.pyplot` on first use rather than at module import.

On the type aliases: rather than making them `TYPE_CHECKING`-only (which would need `from __future__ import annotations` across ~140 consumer modules), I'll keep runtime `object` fallbacks so the existing annotations still resolve, while type checkers see the real matplotlib types. That keeps the diff inside `plot.py` instead of touching every metric.

Measured locally on current `master` (Python 3.13, matplotlib and scipy installed, torchvision not):

```
import torchmetrics : median 5559 ms
sys.modules         : 2158
heavy loaded        : scipy.signal, matplotlib.pyplot
```

I'll report the same numbers after the change, and add a regression test asserting neither module is in `sys.modules` after a bare `import torchmetrics`, so this doesn't silently come back.
